torch/nn/modules/activation.py

```diff
@@ -1062,7 +1062,12 @@ def forward(self, query: Tensor, key: Tensor, value: Tensor, key_padding_mask: O
         if not is_batched:
             why_not_fast_path = f"input not batched; expected query.dim() of 3 but got {query.dim()}"
         elif query is not key or key is not value:
+            # When lifting this restriction, don't forget to either
+            # enforce that the dtypes all match or test cases where
+            # they don't!
             why_not_fast_path = "non-self attention was used (query, key, and value are not the same Tensor)"
+        elif query.dtype != self.in_proj_bias.dtype:
+            why_not_fast_path = f"dtypes of query ({query.dtype}) and self.in_proj_bias ({self.in_proj_bias.dtype}) don't match"
```

Comment:
we could convert the bias to match and warn about the perf impact if this is somehow a reasonable thing people could do

Reply:
Uh oh, we might have missed something big. What if [self.in_proj_bias is None](https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/activation.py#L972)?